.github/workflows/tier_one_coverage.yml

```diff
@@ -0,0 +1,77 @@
+name: TierOneCoverage
+
+on: workflow_dispatch
+
+jobs:
+  check-coverage:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.9'
+
+      - name: Make Temp Dir for Artifacts
+        run: |
+          pwd
+          mkdir coverageDir 
+          touch coverageDir/coverage.csv
+
+      - name: Run Coverage on Tier One Plugins
+        env:
+          TIER_ONE_PLUGINS_ICON: math active_directory_ldap advanced_regex microsoft_atp_safe_links base64 string ssh basename subnet microsoft_teams csv python_3_script type_converter rapid7_insight_agent datetime rapid7_insightidr whois dig rapid7_insightvm rapid7_intsights extractit rapid7_vulndb rapid7_insightappsec hashit rest rapid7_insights html servicenow jira sleep
+
+        run: |
+          set +e
+          cd plugins
+          for plugin in $TIER_ONE_PLUGINS_ICON
+          do
+            cd $plugin
+            echo $plugin
+            workingDir=$(pwd)
+            
+            # Creating venv so we can install required libraries and later delete them to avoid conflicts
+            python3 -m venv venv
+            source venv/bin/activate
+          
+            # Installing previously mentioned libraries
+            pip3 install setuptools==41.2.0 --no-cache-dir --disable-pip-version-check
```

Comment:
That version `41.2.0` is strict or we can go for `--upgrade setuptools` here? 😄

Reply:
we chose the 41.2.0 version on purpose as newer version sometimes clashed with other imports.

Reply:
is it worth adding a comment that we need 41.2.0 to stop the 2to3 issue we were seeing? otherwise next time someone works on this it could be changed to download latest and the headache starts all over again